test/micro/org/openjdk/bench/java/lang/CompareInteger.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright (c) 2022, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+package org.openjdk.bench.java.lang;
+
+import org.openjdk.jmh.annotations.Benchmark;
+import org.openjdk.jmh.annotations.BenchmarkMode;
+import org.openjdk.jmh.annotations.Mode;
+import org.openjdk.jmh.annotations.OutputTimeUnit;
+import org.openjdk.jmh.annotations.Scope;
+import org.openjdk.jmh.annotations.Setup;
+import org.openjdk.jmh.annotations.State;
+import org.openjdk.jmh.annotations.Param;
+import java.util.random.RandomGenerator;
+import java.util.random.RandomGeneratorFactory;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Tests for compare() and compareUnsigned() methods in java.lang.Integer
+ */
+@BenchmarkMode(Mode.AverageTime)
+@OutputTimeUnit(TimeUnit.NANOSECONDS)
+@State(Scope.Thread)
+public class CompareInteger {
+
+    RandomGenerator rng;
+
+    @Param({"mixed", "lessThanEqual", "greaterThanEqual", "equal"})
+    String mode;
+
+    @Param({"1024"})
+    int BUFFER_SIZE;
+
+    int[] input1, input2, outputs;
+
+    @Setup
+    public void setup() {
+        input1 = new int[BUFFER_SIZE];
+        input2 = new int[BUFFER_SIZE];
+        outputs =  new int[BUFFER_SIZE];
+        RandomGenerator rng = RandomGeneratorFactory.getDefault().create(0);
+
+        for (int i = 0; i < BUFFER_SIZE; i++) {
+            input1[i] = rng.nextInt();
+            if (mode.equals("equal")) {
+                input2[i] = input1[i];
+                continue;
+            }
+            else input2[i] = rng.nextInt();
+
+            if (!mode.equals("mixed")) {
+                boolean doSwap = (mode.equals("lessThanEqual") && input1[i] > input2[i]) ||
+                                (mode.equals("greaterThanEqual") && input1[i] < input2[i]);
+                if (doSwap) {
+                    int tmp = input1[i];
+                    input1[i] = input2[i];
+                    input2[i] = tmp;
+                }
+            }
```

Comment:
Logic re-organization suggestion:- 
 
```
 for (int i = 0 ; i < BUFFER_SIZE; i++) {
    input1[i] = rng.nextLong();
 }

 if (mode.equals("equals") {
    GRADIANT = 0;
 } else if (mode.equals("greaterThanEquals")) {
    GRADIANT = 1;
 } else {
    assert mode.equals("lessThanEqual");
    GRADIANT = -1;
 }

 for(int i = 0 ; i < BUFFER_SIZE; i++) {
    input2[i] = input1[i] + i*GRADIANT;
 }
```

Reply:
The suggested refactoring is definitely elegant but one rare possibility is overflow due to the addition/subtraction. The swap logic doesn't have that problem.

Reply:
> The suggested refactoring is definitely elegant but one rare possibility is overflow due to the addition/subtraction. The swap logic doesn't have that problem.

Given that BUFFER_SIZE is 1024 can we not circumvent this problem by using rng.nextLong(4096) in initialization loop.